test/framework/docker_logcollector.go

```diff
@@ -76,6 +76,25 @@ func (k DockerLogCollector) CollectMachinePoolLog(ctx context.Context, _ client.
 	return kerrors.NewAggregate(errs)
 }
 
+func (k DockerLogCollector) CollectInfrastructureLogs(ctx context.Context, clusterName string, outputPath string) error {
+	containerRuntime, err := container.NewDockerClient()
+	if err != nil {
+		return err
+	}
+	ctx = container.RuntimeInto(ctx, containerRuntime)
+
+	lbContainerName := clusterName + "-lb"
+
+	f, err := fileOnHost(filepath.Join(outputPath, fmt.Sprintf("%s.log", lbContainerName)))
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+
+	return containerRuntime.ContainerDebugInfo(ctx, lbContainerName, f)
```

Comment:
Looks like the container info is not derreferenced properly:

`{ContainerJSONBase:0xc001242dc0 Mounts:[{Type:bind Name: Source:/lib/modules Destination:/lib/modules Driver: Mode:ro RW:false Propagation:rprivate} {Type:bind Name: Source:/dev/mapper Destination:/dev/mapper Driver: Mode: RW:true Propagation:rprivate} {Type:volume Name:b4fbc82c8c7bbad7f8cb6acf0f8cab66a4780ff67ef88646c2bd8f5ffbbc9ec3 Source:/var/lib/docker/volumes/b4fbc82c8c7bbad7f8cb6acf0f8cab66a4780ff67ef88646c2bd8f5ffbbc9ec3/_data Destination:/var Driver:local Mode: RW:true Propagation:}] Config:0xc000702000 NetworkSettings:0xc000b52000}
`

Can we get the actual struct values for e.g. NetworkSettings?

Reply:
/lgtm remove

Reply:
Is this coming from:
```
	fmt.Fprintln(w, "Inspected the container:")
	fmt.Fprintf(w, "%+v\n", containerInfo)
```
?

I would suggest to just use json.Marshal and print the result